### DISCUSSION TITLE
Create queries programatically

### DISCUSSION THREAD

**Discussion by PhantomCarrot** [Budibase/budibase#14506] [Help] [answered]
Hello!
I have to create, in an automatic way, automations that will use queries, i manage to create those automation easily;
I have also to create those queries but it seems that POST on /api/queries is blocked

Is there a hidden way to create queries in an existing datasource (or import?) 

Thanks!

**Comment by PhantomCarrot** [comment]
No one have ever faced this need ?

**Comment by ConorWebb96** [comment]
Hey @PhantomCarrot,

You should read the below documentation.

- https://docs.budibase.com/docs/external-data-connector
- https://docs.budibase.com/docs/rest-queries
- https://docs.budibase.com/docs/rest-bindings
- https://docs.budibase.com/docs/data-sources#custom-queries

**Reply by PhantomCarrot** [reply]
Hi @ConorWebb96 
Thanks, but i expected more a programmatic approach, my goal is to programmatically create a rest query

Based on one of your anwser to my questions i create the automation and it works well (https://github.com/Budibase/budibase/discussions/13697)
=> /api/automations POST OK
(automation is just a cron trigger and a call api task)

I've issue however to create a queries programmatically , i have a Method Not Allowed
https://xxx.budibase.app/api/queries

Any workaround ?

**Reply by ConorWebb96** [reply, marked as answer]
Thanks for the explanation, I understand now.

Like i said in that answer the API I showcased isn't surfaced publically. Therefore you won't be able to use it with the external data connector. This only allow you to select queries and rest requests you have setup via their relevant data sources.

You won't be able to dynamically create these queries, they will need to be made ahead of time. What you can do, is trigger and search for them using our public API. You can see more on this [here](https://docs.budibase.com/reference/querysearch).

Hopefully this helps.
